Sources/SwiftParser/Diagnostics/LexerDiagnosticMessages.swift

```diff
@@ -0,0 +1,120 @@
+//===--- LexerDiagnosticMessages.swift ------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+@_spi(RawSyntax) import SwiftSyntax
+
+fileprivate let diagnosticDomain: String = "SwiftLexer"
+
+/// A error diagnostic whose ID is determined by the diagnostic's type.
+public protocol LexerError: DiagnosticMessage {
+  var diagnosticID: MessageID { get }
+}
+
+public extension LexerError {
+  static var diagnosticID: MessageID {
+    return MessageID(domain: diagnosticDomain, id: "\(self)")
+  }
+
+  var diagnosticID: MessageID {
+    return Self.diagnosticID
+  }
+
+  var severity: DiagnosticSeverity {
+    return .error
+  }
+}
+
+public protocol LexerNote: NoteMessage {
+  var fixItID: MessageID { get }
+}
+
+public extension LexerNote {
+  static var fixItID: MessageID {
+    return MessageID(domain: diagnosticDomain, id: "\(self)")
+  }
+
+  var fixItID: MessageID {
+    return Self.fixItID
+  }
+}
+
+public protocol LexerFixIt: FixItMessage {
```

Comment:
Are we anticipating to use lexer notes and Fix-Its? If not, I think it’s fine to remove these for now (to keep things simpler) and add them once needed.